src/measure_theory/decomposition/lebesgue.lean

```diff
@@ -574,6 +571,90 @@ instance have_lebesgue_decomposition_of_finite_measure
         add_comm, ennreal.add_sub_cancel_of_le (hle A hA)] },
 end⟩
 
+instance {μ : measure α} {S : μ.finite_spanning_sets_in {s : set α | measurable_set s}} (n : ℕ) :
+  finite_measure (μ.restrict $ S.set n) :=
+⟨by { rw [restrict_apply measurable_set.univ, set.univ_inter], exact S.finite _ }⟩
+
+/-- **The Lebesgue decomposition theorem**: Any pair of σ-finite measures `μ` and `ν`
+`have_lebesgue_decomposition`. That is to say, there exists a measure `ξ` and a measurable function
+`f`, such that `ξ` is mutually singular with respect to `ν` and `μ = ξ + ν.with_density f` -/
+@[priority 99] -- see Note [lower instance priority]
+instance have_lebesgue_decomposition_of_sigma_finite
+  (μ ν : measure α) [sigma_finite μ] [sigma_finite ν] :
+  have_lebesgue_decomposition μ ν :=
+⟨begin
+  obtain ⟨S, T, h₁, h₂⟩ := exists_eq_disjoint_finite_spanning_sets_in μ ν,
```

Comment:
can you either sketch the math proof in a comment at the beginning of the proof, or add short comments along the proof to explain what you are doing?